compiler/rustc_typeck/src/lib.rs

```diff
@@ -57,7 +57,7 @@ This API is completely unstable and subject to change.
 
 #![allow(rustc::potential_query_instability)]
 #![doc(html_root_url = "https://doc.rust-lang.org/nightly/nightly-rustc/")]
-#![feature(bool_to_option)]
+#![cfg_attr(bootstrap, feature(bool_to_option))]
```

Comment:
The compiler is always built with the new standard library, never with the bootstrap standard library. So `cfg(bootstrap)` should only be necessary in `core`/`alloc`/`std`, not in `rustc_*`.

```suggestion
```

Reply:
@m-ou-se *Ah*, thank you. I'd only encountered `bootstrap` used for language features, where it was needed even in the compiler. I hadn't encountered it used for library features. Makes perfect sense, thanks!